src/webhooks/router.py

```diff
@@ -0,0 +1,59 @@
+import logging
+
+from typing import Any
+from webhooks.responses import *
+from urllib.parse import urlparse
+
+from fastapi import APIRouter, Request, Body
+from fastapi import HTTPException
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+router = APIRouter()
+
+POSSIBLE_ORIGINS = {
+    "facebookplatform/1.0 (+http://developers.facebook.com)": whatsapp_serializer,
+    "facebookexternalua": whatsapp_serializer
+}
+
+POSSIBLE_RESPONSES = {
+    "facebookplatform/1.0 (+http://developers.facebook.com)": whatsapp_get_response,
+    "facebookexternalua": whatsapp_post_response
+}
+
+def detect_origin(request):
+    """
```

Comment:
I think that this detection could be more simple, if we just expect the provider identifier on request's path 

like this: 
`/api/webhooks/<provider>`

Reply:
`python-social-auth` uses this solution to handle many providers 

https://github.com/python-social-auth/social-app-django/blob/master/social_django/urls.py#L14